sdks/python/apache_beam/io/external/kafka.py

```diff
@@ -0,0 +1,102 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+from apache_beam import ExternalTransform
+from apache_beam import pvalue
+from apache_beam.coders import BytesCoder
+from apache_beam.coders import IterableCoder
+from apache_beam.coders import TupleCoder
+from apache_beam.coders.coders import LengthPrefixCoder
+from apache_beam.portability.api.external_transforms_pb2 import ConfigValue
+from apache_beam.portability.api.external_transforms_pb2 import ExternalConfigurationPayload
+from apache_beam.transforms import ptransform
+
+
+class ReadFromKafka(ptransform.PTransform):
```

Comment:
Thanks. I think we should also provide end-to-end instructions on how a user can use this transform in a pipeline, including for example, steps for starting up the expansion service (here or elsewhere) given the this is the first real world source that some of our users might try to use out of the box in the next Beam release.

Reply:
I think yes.  Otherwise users will have no way of knowing how to use this in a pipeline. I think the instructions can be as simple as steps to get the expansion service running (or a pointer to a script for doing this) and providing the address to the ReadFromKafka transform (no need to mention runner setup).

I agree that when things are more stable, we should develop a Webpage for cross-language transforms with complete instructions, examples (for transform authors and users), etc. (not in this PR).

Reply:
I've included a pointer to the Flink Job Server which starts the Java expansion service. Additionally, I've linked documentation pages which explain how to start the Flink Job Server. Should be sufficient for now.